diff --git a/src/sound/ym2151.c b/src/sound/ym2151.c
index bae3a69..6311f01 100644
--- a/src/sound/ym2151.c
+++ b/src/sound/ym2151.c
@@ -1425,7 +1425,8 @@ static void ym2151_postload_refresh(void)
 	{
 		for (j=0; j<8; j++)
 		{
-			set_connect(&YMPSG[i].oper[j*4], YMPSG[i].connect[j], j);
+			/* set_connect takes the channel, then the algorithm */
+			set_connect(&YMPSG[i].oper[j*4], j, YMPSG[i].connect[j]);
 		}
 
 		if (YMPSG[i].timer_A_left >= 0)
